src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php

```diff
@@ -131,13 +131,30 @@ private static function addChoiceView($choice, $value, $label, $keys, &$index, $
             $label = false === $dynamicLabel ? false : (string) $dynamicLabel;
         }
 
+        // BC
+        if (is_array($attr)) {
+            if (isset($attr[$key])) {
+                @trigger_error('Passing an array of arrays to the "choice_attr" option with choice keys as keys is deprecated since version 3.2 and will no longer be supported in 4.0. Use a "\Closure" instead.', E_USER_DEPRECATED);
+                $attr = $attr[$key];
+            } else {
+                foreach ($attr as $a) {
+                    if (is_array($a)) {
```

Comment:
The deprecated case where an array is used and you have to nest arrays of attributes by choice indexes. Then one may not define all indexes. But we need to support the new use case too. So if a nested array exists but not for the current choice, we need to define its attributes to an empty array.

Maybe the comment is not explicit enough?